FEXHeaderUtils/FEXHeaderUtils/Filesystem.h

```diff
@@ -14,4 +22,270 @@ namespace FHU::Filesystem {
   inline bool Exists(const char *Path) {
     return access(Path, F_OK) == 0;
   }
+
+  inline bool Exists(const fextl::string &Path) {
+    return access(Path.c_str(), F_OK) == 0;
+  }
+
+  /**
+   * @brief Creates a directory at the provided path.
+   *
+   * @param Path The path to create a directory at.
+   *
+   * @return True if the directory was created or already exists.
+   */
+  inline bool CreateDirectory(const fextl::string &Path) {
+    auto Result = ::mkdir(Path.c_str(), 0777);
+    if (Result == 0) {
+      return true;
+    }
+
+    if (Result == -1 && errno == EEXIST) {
+      // If it exists, we need to check if it is a file or folder.
+      struct stat buf;
+      if (stat(Path.c_str(), &buf) == 0) {
+        // Check to see if the path is a file or folder. Following symlinks.
+        return S_ISDIR(buf.st_mode);
+      }
+    }
+
+    // Couldn't create, or the path that existed wasn't a folder.
+    return false;
+  }
+
+  /**
+   * @brief Creates a directory tree with the provided path.
+   *
+   * @param Path The path to create a tree at.
+   *
+   * @return True if the directory tree was created or already exists.
+   */
+  inline bool CreateDirectories(const fextl::string &Path) {
+    // Try to create the directory initially.
+    if (CreateDirectory(Path)) {
+      return true;
+    }
+
+    // Walk the path in reverse and create paths as we go.
+    fextl::string TmpPath {Path.substr(0, Path.rfind('/', Path.size() - 1))};
+    if (!TmpPath.empty() && CreateDirectories(TmpPath)) {
+      return CreateDirectory(Path);
+    }
+    return false;
+  }
+
+  /**
+   * @brief Extracts the filename component from a file path.
+   *
+   * @param Path The path to create a directory at.
+   *
+   * @return The filename component of the path.
+   */
+  inline fextl::string GetFilename(const fextl::string &Path) {
+    auto LastSeparator = Path.rfind('/');
+    if (LastSeparator == fextl::string::npos) {
+      // No separator. Likely relative `.`, `..`, `<Application Name>`, or empty string.
+      return Path;
+    }
+
+    return Path.substr(LastSeparator + 1);
+  }
+
+  inline fextl::string ParentPath(const fextl::string &Path) {
+    auto LastSeparator = Path.rfind('/');
+    if (LastSeparator == fextl::string::npos) {
+      // No separator. Likely relative `.`, `..`, `<Application Name>`, or empty string.
+      return Path;
+    }
+
+    return Path.substr(0, LastSeparator);
+  }
+
+  inline bool IsRelative(const fextl::string &Path) {
+    return Path.starts_with('/');
+  }
+
+  enum class CopyOptions {
+    NONE,
+    SKIP_EXISTING,
+    OVERWRITE_EXISTING,
+  };
+
+  inline bool CopyFile(const fextl::string &From, const fextl::string &To, CopyOptions Options = CopyOptions::NONE) {
```

Comment:
docstring needed (just referring to `std::filesystem::copy_file` is fine though)

Reply:
Resolved.